monkey/common/types/utils/socket_address.py

```diff
@@ -0,0 +1,18 @@
+from ipaddress import IPv4Address
+
+from common.network.network_utils import address_to_ip_port
+from common.types import SocketAddress
+
+
+def socketaddress_from_string(address_str: str) -> SocketAddress:
```

Comment:
I think this is ultimately constructor, but python doesn't allow you to have overloaded constructors. The "pythonic" (and I use that term loosely) way to overload constructors is with classmethods.

This could be a class method on `SocketAddress`.